store/file/ods.go

```diff
@@ -108,22 +109,50 @@ func writeODSFile(w io.Writer, eds *rsmt2d.ExtendedDataSquare, axisRoots *share.
 		return nil, fmt.Errorf("writing axis roots: %w", err)
 	}
 
-	for _, shr := range eds.FlattenedODS() {
-		if _, err := w.Write(shr); err != nil {
-			return nil, fmt.Errorf("writing shares: %w", err)
-		}
+	// write quadrants
+	err = writeQuadrant(w, eds, int(h.shareSize), 0)
+	if err != nil {
+		return nil, fmt.Errorf("writing Q1: %w", err)
 	}
 	return h, nil
 }
 
+// writeQuadrant writes the quadrant of the square to the writer. it writes the quadrant in row-major
+// order. It uses buffer to write the shares in bulk (row by row), which improves the write performance.
+func writeQuadrant(w io.Writer, eds *rsmt2d.ExtendedDataSquare, shareSize, quadrantIdx int) error {
+	fromRow := quadrantIdx / 2 * int(eds.Width()) / 2
+	toRow := fromRow + int(eds.Width())/2
+
+	fromCol := quadrantIdx % 2 * int(eds.Width()) / 2
+	toCol := fromCol + int(eds.Width())/2
+
+	buf := bufio.NewWriterSize(w, shareSize*int(eds.Width()))
+	for rowIdx := fromRow; rowIdx < toRow; rowIdx++ {
+		for colIdx := fromCol; colIdx < toCol; colIdx++ {
+			share := eds.GetCell(uint(rowIdx), uint(colIdx))
+			_, err := buf.Write(share)
+			if err != nil {
+				return fmt.Errorf("writing shares: %w", err)
+			}
+		}
+	}
+	err := buf.Flush()
+	if err != nil {
+		return fmt.Errorf("flushing buffer: %w", err)
+	}
+	return nil
+}
+
 func writeAxisRoots(w io.Writer, roots *share.AxisRoots) error {
+	buf := make([]byte, 0, share.AxisRootSize*len(roots.RowRoots))
```

Comment:
TBH I don't follow why allocating buffer, filling it with data and then writing to `io.Writer` is faster/better. Feels like the most improvement comes from `writeQuadrant` which uses `bufio` rather then this func. WDYT ?

Reply:
Thank you. Fixed in recent comments

Reply:
We improved that logic in sync with Vlad on his repo and merged in here